examples/src/components/CopyLink.tsx

```diff
@@ -0,0 +1,33 @@
+'use client'
+import { useState } from 'react'
+
+export default function CopyLink() {
+  const [buttonState, setButtonState] = useState('Copy link')
```

Comment:
nit: rather than setting UI text in state, I would prefer here to do something like:

```tsx
const [copied, setCopied] = useState(false)
```

and then when rendering the button

```tsx
{clicked ? 'Copy link' : 'Copied!'}
```

In this case the component is simple enough that it doesn't make a big difference, but this decouples the state from the copy which makes some things easier, like if we wanted to translate the UI down the road.